tools/outline_proxy_controller/outline_controller_server.h

```diff
@@ -47,22 +51,49 @@ class OutlineClientSession : public std::enable_shared_from_this<OutlineClientSe
 
 public:
   /**
-   * @brief Start serving requests from a specific Outline client asynchronously.
+   * @brief Start a session for a specific Outline client asynchronously.
    *
    * @return boost::asio::awaitable<void> A co_awaitable C++20 coroutine.
    */
   boost::asio::awaitable<void> Start();
 
 private:
   /**
-   * @brief Execution result of a client request command.
+   * @brief Start serving requests from a specific Outline client asynchronously.
    */
+  boost::asio::awaitable<void> ServeClientCommands();
+
+  /**
+   * @brief Start monitoring network changes and update connection status asynchronously.
+   */
+  boost::asio::awaitable<void> MonitorNetworkChanges();
+
+private:
+  /** @brief `TunnelStatus` in "src/www/app/tunnel.ts" */
+  enum class ConnectionState : int {
+    kConnected = 0,
+    kDisconnected = 1,
+    kReconnecting = 2,
+  };
+
+  /** @brief Execution result of a client request command. */
   struct CommandResult {
-    int status;
-    std::string result;
+    int status_code;
+    std::optional<std::string> error_message;
     std::string action;
+    std::optional<ConnectionState> connection_state;
   };
 
+  /** @brief Factory method to create a success result. */
+  static CommandResult SucceededResult(const std::string &action);
```

Comment:
If these are private and static, why not define as standalone functions in the .cpp?

Reply:
Because `CommandResult` is a private struct defined in `OutlineClientSession` class, and it is also the return type of instance method `RunClientCommand()`, so we cannot move the definition and these static methods to .cpp file.

Or we can sacrifice a little bit of encapsulation, make `CommandResult` a public struct in .h file, then we will be able to define and implement these methods completely in .cpp file.